### PR TITLE
PDG: Move `Mass` method in the header

### PR DESCRIPTION
**Pull request by vkucera**
Requires https://github.com/AliceO2Group/O2Physics/pull/3104

### CODE REVIEW

**Comment by vkucera** [comment]
@ktf @ddobrigk @benedikt-voelkel Please have a look.

**Comment by vkucera** [comment]
Preparation in O2Physics merged. Ready to go!

File: Framework/PhysicsSupport/include/Framework/O2DatabasePDGPlugin.h [hunk 1]
```diff
@@ -8,17 +8,27 @@
...
-
 struct O2DatabasePDGImpl : public TDatabasePDG {
-  Double_t Mass(int pdg, bool& success);
+  Double_t Mass(int pdg)
+  {
+    // wrap our own Mass function to expose it in the service
+    bool success = false;
+    auto mass = o2::O2DatabasePDG::Mass(pdg, success, this);
+    if (!success) {
+      LOGF(error, "Unknown particle with PDG code %d", pdg);
```
Comment (vkucera) [inline Framework/PhysicsSupport/include/Framework/O2DatabasePDGPlugin.h:28]: Should we throw an error or a warning?

**Comment by benedikt-voelkel** [comment]
@ktf I would be fine with these changes. They do not touch any internals of `SimulationDataFormats` so I reckon the decision would be up to you since it is mostly your code that is touched.

**Pull request merged**